### PR TITLE
Bump operator-sdk version

### LINKED ISSUE #345: Bump operator-sdk version
## Feature Request

Project development requires usage higher version of operator-sdk and kubernetes.

**Describe the solution you'd like to see**
Update operator-sdk to 0.19.x and go to 1.16, unlocking following updates.

### PR DESCRIPTION
**Pull request by AKamyshnikova**
Update operator-sdk to 0.19.4 and related k8s.io modules
to v0.19.13.

Bump go version to 1.16

Fixes #345

Signed-off-by: akamyshnikova <akamyshnikova@mirantis.com>

| Q               | A
| --------------- | ---
| Bug fix?        | []
| New feature?    | [X]
| API breaks?     | []
| Deprecations?   | []
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | Apache 2.0

### Checklist

- [X ] Implementation tested
- [ ] Logging code meets the guideline
- [ ] User guide and development docs updated (if needed)
- [ ] Append changelog

### CODE REVIEW

**Comment by cscetbon** [comment]
@AKamyshnikova can you rebase your change on master ? When it's done I'll trigger all the end to end tests to confirm it doesn't break anything

**Comment by AKamyshnikova** [comment]
@cscetbon Done

**Comment by cscetbon** [comment]
@AKamyshnikova you need to replace https://github.com/Orange-OpenSource/casskop/blob/8f63813343b3cbd19ac8c6ea6008d722a16ed04a/Makefile#L119 as well. Probably the image could be built in CircleCI as well. We had to use https://github.com/kubernetes-sigs/controller-tools/compare/master...erdrix:v0.3.0-forked_pr317 and that's what uses our build image, however I think it's be merged in the main repo so we should be good now.

**Comment by AKamyshnikova** [comment]
@cscetbon What would be the best way to do this? Should I use your forked operator-sdk repo to build and upload new version of operator-sdk image?

**Comment by AKamyshnikova** [comment]
I've updated code, but I don't have rights to upload new ci image.

**Comment by cscetbon** [comment]
@AKamyshnikova it fails with the new version of the build image, see https://app.circleci.com/pipelines/github/Orange-OpenSource/casskop/1636/workflows/dc14d951-ac39-4e2f-9fd1-3df55c08852d/jobs/16559. Still relates to the fix we applied. You can probably look into it by trying locally ?

**Comment by AKamyshnikova** [comment]
@cscetbon sure, will look into this. Seems that required fix is only since controller-tools 0.4.0 https://github.com/kubernetes-sigs/controller-tools/pull/317/commits/71b6e916ad0a7ea1008bd00c679c6ccba893ccca 
Did you try using controller-gen directly for crd generation? I can try this approach if it will be accepted. Or I can make similar for fork for 19.x operator-sdk branch.

**Comment by cscetbon** [comment]
> @cscetbon sure, will look into this. Seems that required fix is only since controller-tools 0.4.0 [kubernetes-sigs/controller-tools@71b6e91](https://github.com/kubernetes-sigs/controller-tools/commit/71b6e916ad0a7ea1008bd00c679c6ccba893ccca)
> Did you try using controller-gen directly for crd generation? I can try this approach if it will be accepted. Or I can make similar for fork for 19.x operator-sdk branch.

We didn't try but if it works, go for it 👍

**Comment by AKamyshnikova** [comment]
I've updated change using controller-gen directly, CRDs fields got reordered, but everything seems to be in place.

File: docker/circleci/Dockerfile [hunk 1]
```diff
@@ -64,6 +64,7 @@ RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | INSTAL
 RUN curl -sSL https://github.com/gotestyourself/gotestsum/releases/download/v0.3.4/gotestsum_0.3.4_linux_amd64.tar.gz | \
   tar -xz -C /usr/local/bin gotestsum
 RUN go get -u golang.org/x/lint/golint
+RUN go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.0
```
Comment (cscetbon) [inline docker/circleci/Dockerfile:67]: 👍

**Review by cscetbon — COMMENTED** [review]
I left one comment and also asked if you can just double check regarding those CRDs changes. It seems to me it's just a reordering but please confirm it. Other than that it looks good

File: Makefile [hunk 2]
```diff
@@ -154,13 +154,35 @@ helm-package:
 	mv cassandra-operator-$(HELM_VERSION).tgz $(HELM_TARGET_DIR)
 	helm repo index $(HELM_TARGET_DIR)/
 
+controller-gen:
+ifeq (, $(shell which controller-gen))
```
[the remark this answers is not in the document]
Comment (AKamyshnikova) [reply Makefile:158]: I used script that was suggested on https://github.com/operator-framework/operator-sdk/issues/2485#issuecomment-676803137

**Comment by AKamyshnikova** [comment]
> I left one comment and also asked if you can just double check regarding those CRDs changes. It seems to me it's just a reordering but please confirm it. Other than that it looks good

I double check CRD changes - just reorder. Thanks!

**Review by cscetbon — DISMISSED** [review]
Thanks for your contribution !

File: Makefile [hunk 3]
```diff
@@ -186,8 +208,10 @@ docker-generate-crds:
 	docker run --rm -v $(PWD):$(WORKDIR) -v $(GOPATH)/pkg/mod:/go/pkg/mod:delegated \
 		-v $(shell go env GOCACHE):/root/.cache/go-build:delegated --env GO111MODULE=on \
 		--env https_proxy=$(https_proxy) --env http_proxy=$(http_proxy) \
-		$(BUILD_IMAGE):$(OPERATOR_SDK_VERSION) /bin/bash -c 'operator-sdk generate crds'
+		orangeopensource/casskop-build:v0.19.4 /bin/bash -c 'controller-gen crd paths=./pkg/apis/... output:dir=./deploy/crds schemapatch:manifests=./deploy/crds'
```
Comment (cscetbon) [inline Makefile:211]: @AKamyshnikova can you change it back ? The version was updated. Maybe you changed that for your tests?

Comment (AKamyshnikova) [reply Makefile]: Oops, sorry this fix this.

**Review by cscetbon — APPROVED** [review]

**Pull request merged**